kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/status/NMStatusConverter.java

```diff
@@ -60,81 +61,109 @@ public class NMStatusConverter {
 
-        NMDeviceState deviceState = NMDeviceState
-                .fromUInt32(deviceProperties.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, NM_DEVICE_PROPERTY_STATE));
+        NMDeviceState deviceState = NMDeviceState.fromUInt32(
+                devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, NM_DEVICE_PROPERTY_STATE));
         builder.withState(deviceStateConvert(deviceState));
 
         builder.withUsbNetDevice(usbNetDevice);
 
-        setDeviceStatus(builder, deviceProperties.getDeviceProperties());
+        setDeviceStatus(builder, devicePropertiesWrapper);
         setIP4Status(builder, ip4configProperties);
-        setWifiStatus(builder, deviceProperties.getWirelessDeviceProperties(),
+        setWifiStatus(builder, devicePropertiesWrapper.getDeviceSpecificProperties().get(),
                 accessPointsProperties.getActiveAccessPoint(), accessPointsProperties.getAvailableAccessPoints(),
                 supportedChannelsProperties.getCountryCode(), supportedChannelsProperties.getSupportedChannels());
 
         return builder.build();
     }
 
-    private static void setDeviceStatus(NetworkInterfaceStatusBuilder<?> builder, Properties deviceProperties) {
-        builder.withAutoConnect(deviceProperties.Get(NM_DEVICE_BUS_NAME, "Autoconnect"));
-        builder.withFirmwareVersion(deviceProperties.Get(NM_DEVICE_BUS_NAME, "FirmwareVersion"));
-        builder.withDriver(deviceProperties.Get(NM_DEVICE_BUS_NAME, "Driver"));
-        builder.withDriverVersion(deviceProperties.Get(NM_DEVICE_BUS_NAME, "DriverVersion"));
+    private static void setDeviceStatus(NetworkInterfaceStatusBuilder<?> builder,
+            DevicePropertiesWrapper devicePropertiesWrapper) {
+        builder.withAutoConnect(devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "Autoconnect"));
+        builder.withFirmwareVersion(
+                devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "FirmwareVersion"));
+        builder.withDriver(devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "Driver"));
+        builder.withDriverVersion(
+                devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "DriverVersion"));
 
-        UInt32 mtu = deviceProperties.Get(NM_DEVICE_BUS_NAME, "Mtu");
+        UInt32 mtu = devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "Mtu");
         builder.withMtu(mtu.intValue());
 
-        String hwAddress = deviceProperties.Get(NM_DEVICE_BUS_NAME, "HwAddress");
-        builder.withHardwareAddress(getMacAddressBytes(hwAddress));
+        try {
+            String hwAddress = devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, NM_HW_ADDRESS);
+            builder.withHardwareAddress(getMacAddressBytes(hwAddress));
+        } catch (DBusExecutionException e) {
+            logger.debug("NetworkManager version equal or lower then 1.24 detected.");
```

Comment:
This is not completely correct: the property is not available for NM versions <1.24. It is available for 1.24!

```suggestion
            logger.debug("NetworkManager version lower then 1.24 detected.");
```